### DISCUSSION TITLE
IngressRouteUDP

### DISCUSSION THREAD

**Discussion by jsmith-dev** [k3s-io/k3s#8567] [Q&A] [answered]
If I would like to expose CoreDNS as a public DNS server - how might I go about enabling this?

The manifest I thought I would like to use (/var/lib/rancher/k3s/server/manifests/dns-ingress.yaml) might look something like this:

---
apiVersion: traefik.io/v1alpha1
kind: IngressRouteUDP
metadata:
  name: dns-ingress
  namespace: kube-system
spec:
  entryPoints:
    - dns
  routes:
  - match: HostSNI(`ns1.mydomain.example`) || HostSNI(`ns2.mydomain.example`)
    services:
    - name: kube-dns
      port: 53
     
However, this results in the following error:

error: resource mapping not found for name: "dns-ingress" namespace: "kube-system" from "/var/lib/rancher/k3s/server/manifests/dns-ingress.yaml": no matches for kind "IngressRouteUDP" in version "traefik.io/v1alpha1"
ensure CRDs are installed first
      
NOTE: If it's too early to use IngressRouteUDP, not possible to enable it, or preferable to route public internet UDP DNS queries to CoreDNS another way... please let me know / point me in the right direction **TYIA**

**Comment by brandond** [comment, marked as answer]
I would just expose it as a LoadBalancer type Service. There is no value in running bare TCP or UDP services through the Ingress controller. The whole point of Ingress is to do TLS offload and/or request routing, and you don't want any of that here.

**Comment by jsmith-dev** [comment]
Thank you @brandond. Also thank you for working on adding the CoreDNS *.override and *.server patterns early this year which facilitate actually using kube-dns in k3s as an outward facing service in the first place. I will adopt the LoadBalance pattern suggested.
